product-scenarios/4-application-development-using-apis/4.1-application-developers-should-be-able-to-use-apis-in-their-applications/4.1.1-registering-an-application/src/test/java/org/wso2/am/scenario/tests/register/application/ApplicationCreationTestCases.java

```diff
@@ -167,54 +150,129 @@ public void validateApplicationWithMandatoryAndOptionsValues(String applicationN
+//        String responseString = apiStore.generateApplicationKey(appKeyRequestGenerator).getData();
+//        JSONObject responseStringJson = new JSONObject(responseString);
+//        assertFalse(responseStringJson.getBoolean(ERROR),
+//                ERROR_GENERATING_PRODUCTION_KEY + APPLICATION_KEY_GENERATION);
+//        assertEquals(responseStringJson.getJSONObject(DATA).getJSONObject(KEY).getString(KEY_STATE), STATUS_APPROVED,
+//                ERROR_GENERATING_PRODUCTION_KEY + APPLICATION_KEY_GENERATION);
+//        assertEquals(new JSONObject(responseStringJson.getJSONObject(DATA).getJSONObject(KEY).getString(APP_DETAILS))
+//                .get(KEY_TYPE), PRODUCTION, ERROR_GENERATING_PRODUCTION_KEY + APPLICATION_KEY_GENERATION);
+//    }
+//
+//    @Test(description = "4.1.1.4", dependsOnMethods = {"testCreateApplicationForKeyGeneration"})
+//    public void testGenerateSandboxKeysForApplication() throws Exception {
+//        APPKeyRequestGenerator appKeyRequestGenerator = new APPKeyRequestGenerator(APPLICATION_KEY_GENERATION);
+//        appKeyRequestGenerator.setKeyType(SANDBOX);
+//        String responseString = apiStore.generateApplicationKey(appKeyRequestGenerator).getData();
+//        JSONObject responseStringJson = new JSONObject(responseString);
+//        assertFalse(responseStringJson.getBoolean(ERROR),
+//                ERROR_GENERATING_PRODUCTION_KEY + APPLICATION_KEY_GENERATION);
+//        assertEquals(responseStringJson.getJSONObject(DATA).getJSONObject(KEY).getString(KEY_STATE), STATUS_APPROVED,
+//                ERROR_GENERATING_SANDBOX_KEY + APPLICATION_KEY_GENERATION);
+//        assertEquals(new JSONObject(responseStringJson.getJSONObject(DATA).getJSONObject(KEY).getString(APP_DETAILS))
+//                .get(KEY_TYPE), SANDBOX, ERROR_GENERATING_SANDBOX_KEY + APPLICATION_KEY_GENERATION);
+//    }
+//
+//    @Test
+//    private void testCreateApplicationForKeyGeneration() throws Exception {
+//        HttpResponse addApplicationResponse = apiStore
+//                .addApplication(URLEncoder.encode(APPLICATION_KEY_GENERATION, UTF_8),
+//                        URLEncoder.encode(APIMIntegrationConstants.APPLICATION_TIER.UNLIMITED, UTF_8),
+//                        "", URLEncoder.encode(APPLICATION_DESCRIPTION, UTF_8));
+//        applicationsList.add(APPLICATION_KEY_GENERATION);
+//        assertEquals(addApplicationResponse.getResponseCode(), Response.Status.OK.getStatusCode(),
+//                ERROR_APPLICATION_CREATION_FAILED + APPLICATION_KEY_GENERATION);
+//        JSONObject addApplicationJsonObject = new JSONObject(addApplicationResponse.getData());
+//        assertFalse(addApplicationJsonObject.getBoolean(ERROR),
+//                ERROR_APPLICATION_CREATION_FAILED + APPLICATION_KEY_GENERATION);
+//        assertEquals(addApplicationJsonObject.get(STATUS), STATUS_APPROVED,
+//                ERROR_APPLICATION_CREATION_FAILED + APPLICATION_KEY_GENERATION);
+//    }
```

Comment:
Please remove these commented tests if they are not needed anymore.

Reply:
This has been fixed with latter PRs.